### PR TITLE
Added script to check OnAllowSDLFunctionaluty with allowed=false and …

### PR DESCRIPTION
**Pull request by vvvakulenko**
…with device parameter

### CODE REVIEW

**Comment by vvvakulenko** [comment]
@istoimenova please review

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+require('user_modules/AppTypes')
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeleteLogsFileAndPolicyTable()
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:RegisterApp_allowed_false_without_device()
+   -- HMI -> SDL: SDL.ActivateApp
+  local RequestId = self.hmiConnection:SendRequest("SDL.ActivateApp", {appID = self.HMIAppID})
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko self.HMIAppID is usually assigned at RAI. In the script you are using "connecttest", so you can use:
appID = self.applications[config.application1.registerAppInterfaceParams.appName]

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  -- SDL -> HMI: SDL.ActivateApp {isSDLAllowed: false, device is omitted}
+  EXPECT_HMIRESPONSE(RequestId,{isSDLAllowed = false, method = "SDL.ActivateApp"})
+    -- HMI -> SDL: SDL.GetUserFriendlymessage {messageCodes: DataConsent}
+    local RequestId = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage", 
+      {language = "EN-US", messageCodes = {"DataConsent"}})
+    -- SDL -> HMI: SDL.GetUserFriendlymessage {messages}
+    EXPECT_HMIRESPONSE(RequestId,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+      -- HMI -> SDL: SDL.OnAllowSDLFunctionality {allowed: false, device, source}
+      self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", 
+      {allowed = false, source = "GUI", device = {id = config.deviceMAC, name = "127.0.0.1"}})
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko In parameter 'device' you should have: isSDLAllowed
See [Policies] Treating the device as consented:
"Policies Manager treats the device as consented: SDL->HMI: SDL.ActivateApp(isSDLAllowed: true, params)"
As result in future 2 TCs should be prepared.

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      -- SDL -> MOB: OnPermissionChange
+      self.mobileSession:ExpectNotification("OnPermissionsChange", {})
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko I didn't manage to find the requirement that SDL should send OnPermissionsChange to mobile. Can you specify, please.

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      -- SDL -> HMI: BC.ActivateApp {level:<"default_hmi" from "pre_DataConsent" section>, params}
+      EXPECT_HMICALL("BasicCommunication.ActivateApp")
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko Should be checked appropriate parameters:
For this requirement at least level + mandatory appID

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        :Do(function(_,data)
+          --hmi side: sending BasicCommunication.ActivateApp response
+          self.hmiConnection:SendResponse(data.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko Please clarify what is expected result code when app stays in NONE level

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        end)
+  -- SDL -> MOB: OnHMIStatus {HMILevel:<"defaultHMI" from "pre_DataConsent" section>, params}
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "NONE", systemContext = "MAIN"}) 
+end
+--check if device_id was updated in local PT
+function Test:CheckValueOfDeviseID()
+device_id = get_id_value(self)
+  print (device_id)  
+  if (device_id == 0) then
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko The check will be better to be done for device_id equal to config.deviceMAC

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    self:FailTestCase("device_id in database was not updated")
+  end
+end
+
+--[[ Postconditions ]]
+commonFunctions:SDLForceStop()
```
Comment (istoimenova) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @vvvakulenko This function is not declared as Test. So you should call it as Test. As example:
commonFunctions:newTestCasesGroup("Postconditions")
function Test:Postcondition_Force_Stop_SDL()
commonFunctions:SDLForceStop(self)
end

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 8]
```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------
+-- Description:
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 9]
```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------
+-- Description: 
+--     1. Preconditions: App is registered
+--     2. Steps: Activate App, send SDL.OnAllowSDLFunctionality with 'allowed=false' and with 'device' to HMI
+-- Requirement summary: 
+--     [Policy] "EXTERNAL_PROPRIETARY" flow: Related HMI API
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+--     [Policies]: OnAllowSDLFunctionality with 'allowed=false' and with 'device' param from HMI
+--
+-- Expected result:
+--     In case PoliciesManager receives SDL.OnAllowSDLFunctionality with 'allowed=false' and with 'device' param, PoliciesManager must record the named 
+--     device ('device' param) as NOT consented in Local PT ("user_consent_records"-> "device" sub-section) and send BasicCommunication.ActivateApp with 
+--     'level' param of the value from 'default_hmi' key of 'pre-DataConsent'section of Local PT to HMI. App should stay in NONE HMI level
+---------------------------------------------------------------------------------------------
+--[[ General Settings for configuration ]]
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+Test = require('connecttest') 
+require('cardinalities')
+require('mobile_session')
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+--[[ Local Functions ]]
+local function get_id_value()
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      return retvalue
+    end
+end
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+require('user_modules/AppTypes')
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeleteLogsFileAndPolicyTable()
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:RegisterApp_allowed_false_without_device()
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua (hunk 1 above)
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+   -- HMI -> SDL: SDL.ActivateApp
+  local RequestId = self.hmiConnection:SendRequest("SDL.ActivateApp", {appID = self.HMIAppID})
+  -- SDL -> HMI: SDL.ActivateApp {isSDLAllowed: false, device is omitted}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+  EXPECT_HMIRESPONSE(RequestId,{isSDLAllowed = false, method = "SDL.ActivateApp"})
+    -- HMI -> SDL: SDL.GetUserFriendlymessage {messageCodes: DataConsent}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    local RequestId = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage", 
+      {language = "EN-US", messageCodes = {"DataConsent"}})
+    -- SDL -> HMI: SDL.GetUserFriendlymessage {messages}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+    EXPECT_HMIRESPONSE(RequestId,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+      -- HMI -> SDL: SDL.OnAllowSDLFunctionality {allowed: false, device, source}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua (hunk 2 above)
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+      self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", 
+      {allowed = false, source = "GUI", device = {id = config.deviceMAC, name = "127.0.0.1"}})
+      -- SDL -> MOB: OnPermissionChange
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua (hunk 3 above)
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+      self.mobileSession:ExpectNotification("OnPermissionsChange", {})
+      -- SDL -> HMI: BC.ActivateApp {level:<"default_hmi" from "pre_DataConsent" section>, params}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua (hunk 4 above)
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+      EXPECT_HMICALL("BasicCommunication.ActivateApp")
+        :Do(function(_,data)
+          --hmi side: sending BasicCommunication.ActivateApp response
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua (hunk 5 above)
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

File: test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+          self.hmiConnection:SendResponse(data.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+        end)
+  -- SDL -> MOB: OnHMIStatus {HMILevel:<"defaultHMI" from "pre_DataConsent" section>, params}
```
Comment (vvvakulenko) [inline test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua]: @istoimenova done in commit fc421843394d33f1f9b522d3294b2a3462852a87

**Comment by istoimenova** [comment]
@LuxoftAKutsan please merge.

**Pull request merged**